File: esbmc.py/Testes02/validador_categoria/validador_somente_letras.py
MAX_LEN: int = 31

def validate_only_letters(category: str, length: int) -> bool:
    i: int = 0
    while i < length:
        assert category[i].isalpha(), \
            "A categoria deve conter apenas letras (sem números)."
        i += 1
    return True


def main() -> None:
    category: str = input() 
    length: int = len(category)
    
    assert length >= 3 and length <= 30

    validate_only_letters(category, length)


main()

File: esbmc.py/Testes02/validador_categoria/validador_somente_letras.cpp
#include <cctype>
#include <cstdlib>

#define MAX_LEN 31

using namespace std;

bool validate_only_letters(const char category[MAX_LEN], int len) {
    for (int i = 0; i < len; i++) {
        __ESBMC_assert(isalpha(category[i]), "A categoria deve conter apenas letras (sem números).");
    }
    return true;
}

int main() {
    char category[MAX_LEN];
    int len;

    __ESBMC_assume(len >= 3 && len <= 30);

    validate_only_letters(category, len);

    return 0;
}
